py/objexcept.c

```diff
@@ -313,7 +313,36 @@ mp_obj_t mp_obj_new_exception_args(const mp_obj_type_t *exc_type, size_t n_args,
 }
 
 mp_obj_t mp_obj_new_exception_msg(const mp_obj_type_t *exc_type, const char *msg) {
-    return mp_obj_new_exception_msg_varg(exc_type, msg);
+    // Check that the given type is an exception type
+    assert(exc_type->make_new == mp_obj_exception_make_new);
+
+    // Try to allocate memory for the message
+    mp_obj_str_t *o_str = m_new_obj_maybe(mp_obj_str_t);
+
+    #if MICROPY_ENABLE_EMERGENCY_EXCEPTION_BUF
+    // If memory allocation failed and there is an emergency buffer then try to use
+    // that buffer to store the string object, reserving room at the start for the
+    // traceback and 1-tuple.
+    if (o_str == NULL
+        && mp_emergency_exception_buf_size >= EMG_TRACEBACK_ALLOC * sizeof(size_t)
```

Comment:
This is of course hard to read and repeats the same (longish) subexpression twice. (And it's 2nd change in row like that, after adding similarly hardish to read code to implement accept queue in modlwip, where no attempt was made to brush it up somehow (hide behind macros/inline funs, whatever)).

Reply:
Indeed reusing this shouldn't be too hard and would do wonders for readability